modules/misc/src/postprocessors/GeneralSensorPostprocessor.C

```diff
@@ -0,0 +1,317 @@
+  return drift_value;
+}
+
+Real GeneralSensorPostprocessor::getEfficiency()
+{
+  Real efficiency_value;
+  // Evaluate drift function at the current time
+  efficiency_value = _efficiency_function.value(_t);
+  return efficiency_value;
+}
+
+Real GeneralSensorPostprocessor::getSignalToNoiseFactor()
+{
+  // Signal to noise factor as a function of time
+  Real signalToNoise_value;
+  signalToNoise_value = _signalToNoise_function.value(_t);
+  return signalToNoise_value;
+}
+
+Real GeneralSensorPostprocessor::getNoiseStdDev()
+{
+  // Noise standard deviation as a function of time
+  Real noise_std_dev;
+  noise_std_dev = _noise_std_dev_function.value(_t);
+  return noise_std_dev;
+}
+
+Real GeneralSensorPostprocessor::getNoise(Real noise_std_dev)
+{
+  // Sample a normal distribution centered around the noise mean, with a width of the noise standard deviation
+  Real noise_value = 0.0;
+  noise_value = _rng.randNormal(0, noise_std_dev);
+  return noise_value;
+}
+
+Real GeneralSensorPostprocessor::getUncertaintyStdDev()
+{
+  // Uncertainty standard deviation as a function of time
+  Real uncertainty_std_dev;
+  uncertainty_std_dev = _uncertainty_std_dev_function.value(_t);
+  return uncertainty_std_dev;
+}
+
+Real GeneralSensorPostprocessor::getUncertainty(Real uncertainty_std_dev)
+{
+  // Uncertainty term
+  Real uncertainty_value;
+  uncertainty_value = _rng.randNormal(0, uncertainty_std_dev);
+  return uncertainty_value;
+}
+
+Real GeneralSensorPostprocessor::getDelay()
+{
+  // delay (tau) as a function of time
+  Real delay_value = 0.0;
+  delay_value = _delay_function.value(_t);
+  return delay_value;
+}
+
+Real GeneralSensorPostprocessor::getDelayedInputSignal(Real delay_value, std::vector<Real> time_values, std::vector<Real> input_signal_values)
+{ 
+  // delayed input signal
+  Real t_desired = _t - delay_value;
+  Real input_signal_delayed;
+
+  if (t_desired < time_values[0])
+    input_signal_delayed = 0;
+
+  else if (t_desired == time_values[0])
+    input_signal_delayed = _input_signal;
+
+  // linear interpolation
+  else if (t_desired > time_values[0] && t_desired <= _t && t_desired >= _t - _dt)
+    input_signal_delayed = _input_signal + (t_desired - _t) * (_pp_old - _input_signal) / (-_dt);
+
+  else if (t_desired > time_values[0] && t_desired < _t - _dt) //&& floor(t_desired/_dt)*_dt == time_values[0]
+  {
+    LinearInterpolation time_and_input_signal(time_values, input_signal_values);
+    input_signal_delayed = time_and_input_signal.sample(t_desired);
+  }
+
+  else
+    cout << "Some error in interpolation values." <<endl;
+
+  return input_signal_delayed;
+}
+
+Real GeneralSensorPostprocessor::getIntegral(std::vector<Real> integrand, std::vector<Real> time_values, Real delay_value)
+{
+  Real integration_value;
+  std::vector<Real> R_function_values;
+  // computing vector of R function values
+  for (size_t i = 0; i < time_values.size(); ++i) 
+      R_function_values.push_back(_R_function.value(_t)); 

```

Comment:
Are you sure this is what you want? The compiler should have warned you that `i` is being unused. Did you mean to say `time_values[i]` instead of `_t` ? Again we_always_ explicitly specify the `std` namespace! So this should be `std::size_t`, better yet, if my first comment is correct:

```
for (const auto t : time_values) 
  R_function_values.push_back(_R_function.value(t)); 
```

use a range based for loop